src/H5FDros3.h

```diff
@@ -70,18 +70,20 @@
  *
  ****************************************************************************/
 
-#define H5FD_CURR_ROS3_FAPL_T_VERSION 1
+#define H5FD_CURR_ROS3_FAPL_T_VERSION 2
 
-#define H5FD_ROS3_MAX_REGION_LEN     32
-#define H5FD_ROS3_MAX_SECRET_ID_LEN  128
-#define H5FD_ROS3_MAX_SECRET_KEY_LEN 128
+#define H5FD_ROS3_MAX_REGION_LEN        32
+#define H5FD_ROS3_MAX_SECRET_ID_LEN     128
+#define H5FD_ROS3_MAX_SECRET_KEY_LEN    128
+#define H5FD_ROS3_MAX_SESSION_TOKEN_LEN 4096
 
 typedef struct H5FD_ros3_fapl_t {
     int32_t version;
     hbool_t authenticate;
     char    aws_region[H5FD_ROS3_MAX_REGION_LEN + 1];
     char    secret_id[H5FD_ROS3_MAX_SECRET_ID_LEN + 1];
     char    secret_key[H5FD_ROS3_MAX_SECRET_KEY_LEN + 1];
+    char    session_token[H5FD_ROS3_MAX_SESSION_TOKEN_LEN + 1];
```

Comment:
Changing a struct in a public header breaks binary compatibility so this change could not be merged downstream to 1.14, etc.

Reply:
the change comes with a bump in `H5FD_CURR_ROS3_FAPL_T_VERSION`, so it is easy to detect clients that have not been recompiled against the updated headers/library. we should be able to restore binary compatibility. shall we walk through the use cases, or are you guys happy to say this is a 1.15 fix?